src/sp/it/pl/util/dev/Fail.kt

```diff
@@ -0,0 +1,38 @@
+@file:JvmName("Fail")
+@file:Suppress("unused")
+
+package sp.it.pl.util.dev
+
+import javafx.application.Platform
+import java.lang.reflect.Field
+import java.lang.reflect.Modifier
+
+
+inline fun fail(message: () -> String? = { null }): Nothing = throw AssertionError(message())
+
+@JvmOverloads
+inline fun failIf(condition: Boolean, message: () -> String = { "" }) {
+    if (condition) fail { "Requirement condition not met: ${message()}" }
+}
+
+inline fun failIfNot(condition: Boolean, message: () -> String) = failIf(!condition, message)
+
+fun <T> noNull(o: T?) = o.noNull()
+
+inline fun <T> T?.noNull(message: () -> String = { "" }): T = this ?: fail { "Null forbidden: ${message()}" }
```

Comment:
I think we should not make this an extension method. It is for Java only. All uses in Kotlin should be refactored to `?: fail { "..." }`. As Java can not see extensions, theres little reason to clutter API of Any? and burden compiler/checker with more work.